src/node/node.rs

```diff
@@ -104,22 +155,62 @@ impl Node {
         .await;
         // Execute operators scheduled on the current node.
         let channel_manager = Arc::new(std::sync::Mutex::new(channel_manager));
-        for operator_info in graph
+        let local_operators: Vec<_> = graph
             .get_operators()
             .into_iter()
             .filter(|op| op.node_id == self.id)
-        {
+            .collect();
+        let num_local_operators = local_operators.len();
+
+        let (operator_tx, rx_from_operators) = std::sync::mpsc::channel();
+        let mut channels_to_operators = HashMap::new();
+
+        for operator_info in local_operators {
             debug!(
                 self.config.logger,
                 "Executing operator {} on node {}", operator_info.id, operator_info.node_id
             );
             let channel_manager_copy = Arc::clone(&channel_manager);
+            let operator_tx_copy = operator_tx.clone();
+            let (tx, rx) = std::sync::mpsc::channel();
+            channels_to_operators.insert(operator_info.id, tx);
             // Launch the operator as a separate async task.
             tokio::spawn(async move {
-                (operator_info.runner)(channel_manager_copy).execute();
+                (operator_info.runner)(channel_manager_copy, operator_tx_copy, rx).execute();
             });
         }
 
+        // Wait for all operators to finish setting up
```

Comment:
Can you please move the code related to waiting for all operators to initialize, and the code broadcasting to all nodes to separate methods?